packages/accounts-controller/src/AccountsController.ts

```diff
@@ -682,19 +684,35 @@ export class AccountsController extends BaseController<
    */
   async #listNormalAccounts(): Promise<InternalAccount[]> {
     const internalAccounts: InternalAccount[] = [];
-    const { keyrings } = await this.messagingSystem.call(
+    const { keyrings, keyringsMetadata } = await this.messagingSystem.call(
       'KeyringController:getState',
     );
-    for (const keyring of keyrings) {
+
+    for (let keyringIndex = 0; keyringIndex < keyrings.length; keyringIndex++) {
+      const keyring = keyrings[keyringIndex];
       const keyringType = keyring.type;
       if (!isNormalKeyringType(keyringType as KeyringTypes)) {
         // We only consider "normal accounts" here, so keep looping
         continue;
       }
 
-      for (const address of keyring.accounts) {
+      for (
+        let accountIndex = 0;
+        accountIndex < keyring.accounts.length;
+        accountIndex++
+      ) {
+        const address = keyring.accounts[accountIndex];
         const id = getUUIDFromAddressOfNormalAccount(address);
 
+        let options = {};
+
+        if (isHdKeyringType(keyring.type as KeyringTypes)) {
+          options = {
+            entropySource: keyringsMetadata[keyringIndex].id,
+            derivationPath: getDerivationPathForIndex(accountIndex),
```

Comment:
I wonder if re-using the `hdPath` from the keyring itself would be better here? 🤔
- https://github.com/MetaMask/accounts/blob/v47.0.0/packages/keyring-eth-hd/src/hd-keyring.ts#L115

We're not using a different one from the one you have hardcoded here, but still, the source of truth here is the keyring IMO.

Reply:
I would not be opposed to reusing, but that property is not available here since the `keyring` is not of type `HdKeyring` like your link but rather [`KeyringObject` from `KeyringController`](https://github.com/MetaMask/core/blob/6e8000b39ca4f35d6a3ffb1d3b9470d8cae5a13d/packages/keyring-controller/src/KeyringController.ts#L271)